doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,394 @@
+# Solve an equation algebraically
+
+Use SymPy to solve an equation algebraically (symbolically). For example, solving $x^2 = y$ yields $x \in \{-\sqrt{y},\sqrt{y}\}$.
+
+Alternatives to consider:
+- SymPy can also [solve many other types of problems including sets of equations](index.md).
+- Some equations cannot be solved algebraically (either at all or by SymPy), 
+so you may have to {func}`solve your equation numerically <sympy.solvers.solvers.nsolve>` instead.
+
+There are two high-level functions to solve equations, {func}`~.solve` and {func}`~.solveset`.
+Here is a simple example of each:
+
+{func}`~.solve`
+
+```py
+>>> from sympy.abc import x, y
+>>> from sympy import solve
+>>> solution = solve(x ** 2 - y, x, dict=True)
+>>> print(solution)
+[{x: -sqrt(y)}, {x: sqrt(y)}]
+```
+
+{func}`~.solveset`
+
+```py
+>>> from sympy import solveset
+>>> from sympy.abc import x, y
+>>> solution = solveset(x**2 - y, x)
+>>> print(solution)
+{-sqrt(y), sqrt(y)}
+```
+
+Here are recommendations on when to use:
+
+- {func}`~.solve`
+    - You need to programmatically extract components (expressions, or individual symbols or constants) 
+    from the output by traversing it.
+    - You want an explicit solution expression that you can use to substitute into something else.
+    - You want to apply assumptions to symbols (for example, $x$ is real) to determine the 
+    domain of the solutions.
```

Comment:
I don't think this last point really makes sense as a reason to use `solve`. I would probably just remove it. In fact I'm not sure about the first point either as written.

How about:
```
- You want to get explicit symbolic representations of the different values
  a variable could take that would satisfy the equation.
- You want to substitute those explicit solution values into other equations
  or expressions involving the same variable.
```